File: src/main/java/edu/berkeley/bid/bullet/JointSensorState.java
package edu.berkeley.bid.bullet;

import java.io.Serializable;

public class JointSensorState implements Serializable {
    private static final long serialVersionUID = 32106451303123498L;

    public double m_jointPosition;
    public double m_jointVelocity;
    public double []m_jointForceTorque;
    public double m_jointMotorTorque;

    public JointSensorState (double jointPosition, double jointVelocity, double []jointForceTorque, double jointMotorTorque) {
	m_jointPosition = jointPosition;
	m_jointVelocity = jointVelocity;
	m_jointForceTorque = jointForceTorque;
	m_jointMotorTorque = jointMotorTorque;
    }

    public JointSensorState() {
	m_jointForceTorque = new double[6];
    }

    @Override
    public String toString () {
    	return "[" +
	    "jointPosition=" + m_jointPosition + ", " +
	    "jointVelocity=" + m_jointVelocity + ", " +
	    "jointForceTorque=" + m_jointForceTorque + ", " +
	    "jointMotorTorque=" + m_jointMotorTorque + "]";
    }

}

File: src/main/scala/BIDMach/rl/environments/bullet/JointSensorState.scala
package BIDMach.rl.environments.bullet;
import BIDMat.DMat;

@SerialVersionUID(347814368796176834L)
class JointSensorState(jss:edu.berkeley.bid.bullet.JointSensorState) extends Serializable {

    val javaSensorState = jss;
    
    def position:Double = javaSensorState.m_jointPosition;
    def velocity:Double = javaSensorState.m_jointVelocity;
    def motorTorque:Double = javaSensorState.m_jointMotorTorque;

    def position_=(v:Double) = javaSensorState.m_jointPosition = v;
    def velocity_=(v:Double) = javaSensorState.m_jointVelocity = v;
    def motorTorque_=(v:Double) = javaSensorState.m_jointMotorTorque = v;

    val forceTorque:DMat = new DMat(1, 6, javaSensorState.m_jointForceTorque);

    def this() = this(new edu.berkeley.bid.bullet.JointSensorState);

    def unpack() = {
	(position, velocity, forceTorque, motorTorque)
    }
	
    override def toString ():String = {
    	"JointSensorState[position=" + position + ", " +
	"velocity=" + velocity + ", " +
	"forceTorque=" + forceTorque + ", " +
	"motorTorque=" + motorTorque + "]";
    }

}
